### PR TITLE
Enable semantic search

### LINKED ISSUE #19904: Add support for semantic search
This tracks the work to expose the semantic search in our library

### PR DESCRIPTION
**Pull request by Mohit-Chakraborty**

### CODE REVIEW

File: sdk/search/Azure.Search.Documents/src/Models/SearchResults.cs [hunk 1]
```diff
@@ -53,6 +53,10 @@ public class SearchResults<T>
         /// </summary>
         public IDictionary<string, IList<FacetResult>> Facets { get; internal set; }
 
+        /// <summary> The answers query results for the search operation;
+        /// <c>null</c> if the <see cref="SearchOptions.QueryAnswer"/> parameter was not specified or set to <see cref="QueryAnswer.None"/>. </summary>
+        public IDictionary<string, IList<AnswerResult>> Answers { get; internal set; }
```
Comment (Mohit-Chakraborty) [inline sdk/search/Azure.Search.Documents/src/Models/SearchResults.cs:58]: I am working on it. Last I checked, this was also returned as an array. Now, I am not getting this in the response at all.

**Comment by Mohit-Chakraborty** [comment]
cc: @tg-msft

File: sdk/search/Azure.Search.Documents/src/Options/SearchOptions.cs [hunk 2]
```diff
@@ -163,13 +163,75 @@ internal string OrderByRaw
...
+            {
+                if (!QueryAnswer.HasValue)
+                {
+                    return null;
+                }
+
+                string queryAnswerStringValue = QueryAnswer.Value.ToString();
+                if (QueryAnswer.Value == Models.QueryAnswer.Extractive)
+                {
+                    queryAnswerStringValue = $"{Models.QueryAnswer.Extractive}|count-{Math.Min(5, QueryAnswerCount)}";
```
Comment (tg-msft) [inline sdk/search/Azure.Search.Documents/src/Options/SearchOptions.cs:198]: I worry this is too magical, but let's leave it for now and think about it more before GA.

File: sdk/search/Azure.Search.Documents/src/SearchClientOptions.cs [hunk 3]
```diff
@@ -30,21 +30,27 @@ public enum ServiceVersion
...
-        internal const ServiceVersion LatestVersion = ServiceVersion.V2020_06_30;
+        internal const ServiceVersion LatestVersion = ServiceVersion.V2020_06_30_Preview;
 
         /// <summary>
         /// The service version to use when creating continuation tokens that
         /// can be passed between different client libraries.  Changing this
         /// value requires updating <see cref="Azure.Search.Documents.Models.SearchContinuationToken"/>.
         /// </summary>
-        internal const ServiceVersion ContinuationTokenVersion = ServiceVersion.V2020_06_30;
+        internal const ServiceVersion ContinuationTokenVersion = ServiceVersion.V2020_06_30_Preview;
```
Comment (tg-msft) [inline sdk/search/Azure.Search.Documents/src/SearchClientOptions.cs:53]: Revert this - it's a long discussion, but we shouldn't change this unless we change the algorithm.

File: sdk/search/Azure.Search.Documents/src/SearchClientOptions.cs [hunk 4]
```diff
@@ -165,7 +171,7 @@ internal static partial class SearchExtensions
         public static SearchClientOptions.ServiceVersion Validate(this SearchClientOptions.ServiceVersion version) =>
             version switch
             {
-                SearchClientOptions.ServiceVersion.V2020_06_30 => version,
+                SearchClientOptions.ServiceVersion.V2020_06_30_Preview => version,
```
Comment (tg-msft) [inline sdk/search/Azure.Search.Documents/src/SearchClientOptions.cs:174]: You need to add new entries, not replace the old entries.

File: sdk/search/Azure.Search.Documents/src/SearchClientOptions.cs [hunk 5]
```diff
@@ -187,7 +193,7 @@ version switch
         public static string ToVersionString(this SearchClientOptions.ServiceVersion version) =>
             version switch
             {
-                SearchClientOptions.ServiceVersion.V2020_06_30 => "2020-06-30",
+                SearchClientOptions.ServiceVersion.V2020_06_30_Preview => "2020-06-30-Preview",
```
Comment (tg-msft) [inline sdk/search/Azure.Search.Documents/src/SearchClientOptions.cs:196]: You need to add new entries, not replace the old entries.

File: sdk/search/Azure.Search.Documents/api/Azure.Search.Documents.netstandard2.0.cs [hunk 6]
```diff
@@ -127,6 +128,10 @@ public partial class SearchOptions
         public bool? IncludeTotalCount { get { throw null; } set { } }
         public double? MinimumCoverage { get { throw null; } set { } }
         public System.Collections.Generic.IList<string> OrderBy { get { throw null; } }
+        public Azure.Search.Documents.Models.QueryAnswer? QueryAnswer { get { throw null; } set { } }
+        public uint QueryAnswerCount { get { throw null; } set { } }
```
Comment (tg-msft) [inline sdk/search/Azure.Search.Documents/api/Azure.Search.Documents.netstandard2.0.cs:132]: Did you pick `uint` or did code gen?

File: sdk/search/Azure.Search.Documents/api/Azure.Search.Documents.netstandard2.0.cs [hunk 7]
```diff
@@ -127,6 +128,10 @@ public partial class SearchOptions
...
+        public Azure.Search.Documents.Models.QueryLanguage? QueryLanguage { get { throw null; } set { } }
+        public Azure.Search.Documents.Models.QuerySpeller? QuerySpeller { get { throw null; } set { } }
```
Comment (tg-msft) [inline sdk/search/Azure.Search.Documents/api/Azure.Search.Documents.netstandard2.0.cs:134]: I think we'll want to iterate on these property names in ApiView a bit.

**Review by tg-msft — APPROVED** [review]
This looks awesome.  The only things that need to be fixed are the version comments in ClientOptions.

File: sdk/search/Azure.Search.Documents/api/Azure.Search.Documents.netstandard2.0.cs (hunk 6 above)
Comment (Mohit-Chakraborty) [reply sdk/search/Azure.Search.Documents/api/Azure.Search.Documents.netstandard2.0.cs:132]: I did. The valid range is [1,5].

File: sdk/search/Azure.Search.Documents/api/Azure.Search.Documents.netstandard2.0.cs (hunk 7 above)
Comment (Mohit-Chakraborty) [reply sdk/search/Azure.Search.Documents/api/Azure.Search.Documents.netstandard2.0.cs:134]: I tried. <shrug/> 
I think `SearchQueryLanguage`, `SearchQuerySpeller` etc would be better.

**Pull request merged**